cdep/src/main/java/io/cdep/cdep/generator/CMakeGenerator.java

```diff
@@ -319,15 +359,80 @@ private void visit(@NotNull Expression expression) {
               specific.size.toString(),
               specific.sha256));
       if (specific.includePath != null) {
-        append("%starget_include_directories(${target} PRIVATE ", prefix);
-        visit(specific.includePath);
-        append(")\n");
+        if (commonIncludes != null) {
+          commonIncludes.add(specific.includePath);
+        }
+        insertingIndex = sb.length();
+        insertingPrefix = prefix;
       }
 
       for(Expression libraryPath : specific.libraryPaths) {
-        append("%starget_link_libraries(${target} ", prefix);
+        dependsInsert = 0;
+        String fullPath= libraryPath.toString();
+        String libName;
+        if(fullPath.contains(".") && fullPath.contains("/") && fullPath.lastIndexOf(".") - fullPath.lastIndexOf("/") > 0)
```

Comment:
I think if this was a named function it would be  little easier for me to understand